linalg/vector.hpp

```diff
@@ -621,6 +627,41 @@ inline void Vector::SetSize(int s, MemoryType mt)
    data.UseDevice(use_dev);
 }
 
+inline void Vector::Reserve(int res)
+{
+   if (res > Capacity())
+   {
+      Memory<T> p(res, data.GetMemoryType());
```

Comment:
```suggestion
      Memory<real_t> p(res, data.GetMemoryType());
```

Sorry, bad copy-paste earlier from `Array<T>`. This should work, but please check!

Reply:
Sorry for not double-checking!

Reply:
No worries! 😄 My bad for the broken suggestion - I just didn't check myself.